qiskit_ibm_runtime/visualization/plot_zne.py

```diff
@@ -0,0 +1,288 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Functions to visualize :class:`~.EstimatorPubResult` ZNE data."""
+
+from __future__ import annotations
+
+from itertools import product
+from typing import Sequence
+from plotly.colors import sample_colorscale
+from plotly.subplots import make_subplots
+import numpy as np
+import plotly.graph_objects as go
+
+from ..utils.estimator_pub_result import EstimatorPubResult
+
+
+def plot_zne(
+    result: EstimatorPubResult,
+    indices: Sequence[tuple[int, ...]] | None = None,
+    names: Sequence[str] | None = None,
+    n_stds: int = 1,
+    mag_tol: float = 10,
+    std_tol: float = 0.2,
+    height: int = 500,
+    width: int = 1000,
+    n_cols: int = 4,
+    colorscale: str = "Aggrnyl",
+    subplots: bool = False,
```

Comment:
I think we talked about this at some point, writing this his for the sake of open-sourcing the converstation.

Have you considered if `subplots=True` and `subplots=False` should be different functions, e.g. `plot_zne` and `plot_zne_subplots` (names just made up now, they can be improved)?
The main body of the functions are different (wrapped in a big `if .. else ..`), as are the outputs and (I imagine) the default values that you may want to have. And it's true that they rely on similar pre- and post-processing logic, but this can be incorporated into private functions and used in both places.

I imagine that this is something that comes up pretty often in scatter plots in other python libraries.
I'm not sure if there is a standard pattern though

Reply:
Yeah, agreed. I have broken it down into two separate methods. The `subplots=True` case is now `plot_zne` and `subplots=False` is `plot_zne_extrapolators`. Seems natural to want to look at each expectation value on its own rather than having them all on the same plot by default.